source/state_representation/include/state_representation/parameters/Parameter.hpp

```diff
@@ -1,92 +1,101 @@
+   * @param name The name of the parameter
+   * @param value The value of the parameter
    */
   explicit Parameter(const std::string& name, const T& value);
 
   /**
    * @brief Copy constructor
-   * @param parameter the parameter to copy
+   * @param parameter The parameter to copy
    */
-  template <typename U>
+  template<typename U>
   Parameter(const Parameter<U>& parameter);
 
   /**
    * @brief Conversion equality
    */
-  template <typename U>
+  template<typename U>
   Parameter<T>& operator=(const Parameter<U>& parameter);
 
   /**
-   * @brief Getter of the value attribute
-   * @return the value attribute
+   * @brief Getter of the value attribute.
+   * @return The value attribute
    */
   const T& get_value() const;
 
   /**
-   * @brief Getter of the value attribute
-   * @return the value attribute
+   * @brief Getter of the value attribute.
+   * @return The value attribute
    */
   T& get_value();
 
   /**
-   * @brief Setter of the value attribute
-   * @param the new value attribute
+   * @brief Setter of the value attribute.
+   * @param The new value attribute
    */
   virtual void set_value(const T& value);
 
   /**
-   * @brief Overload the ostream operator for printing
-   * @param os the ostream to append the string representing the State to
-   * @param parameter the Parameter to print
-   * @return the appended ostream
+   * @brief Overload the ostream operator for printing.
+   * @param os The ostream to append the string representing the State to
+   * @param parameter The Parameter to print
+   * @return The appended ostream
    */
-  template <typename U>
+  template<typename U>
   friend std::ostream& operator<<(std::ostream& os, const Parameter<U>& parameter);
 };
 
-template <typename T>
-template <typename U>
-Parameter<T>::Parameter(const Parameter<U>& parameter) : ParameterInterface(parameter), value(parameter.get_value()) {}
+template<typename T>
+template<typename U>
+Parameter<T>::Parameter(const Parameter<U>& parameter) : ParameterInterface(parameter), value_(parameter.get_value()) {}
 
-template <typename T>
-template <typename U>
+template<typename T>
+template<typename U>
 Parameter<T>& Parameter<T>::operator=(const Parameter<U>& parameter) {
   Parameter<T> temp(parameter);
   *this = temp;
   return *this;
 }
 
-template <typename T>
+template<typename T>
 inline const T& Parameter<T>::get_value() const {
-  return this->value;
+  return this->value_;
 }
 
-template <typename T>
+template<typename T>
 inline T& Parameter<T>::get_value() {
-  return this->value;
+  return this->value_;
 }
 
-template <typename T>
+template<typename T>
 inline void Parameter<T>::set_value(const T& value) {
   this->set_filled();
-  this->value = value;
+  this->value_ = value;
+}
+
+template<typename T>
+static std::shared_ptr<Parameter<T>> create_parameter_ptr(const std::string& name, const T& param_value) {
+  return std::make_shared<Parameter<T>>(Parameter<T>(name, param_value));
```

Comment:
I'm pretty sure you can just pass the constructor arguments directly to `make_shared` without having to "construct in place". Maybe I'm wrong through because of the template deduction. Could you try it?
```suggestion
  return std::make_shared<Parameter<T>>(name, param_value);
```
`std::make_shared<Parameter<T>>(Parameter<T>(name, param_value))`

Reply:
Yes you are right, that works too! This reduces the contribution value of this PR, but I still like my helper class :smile: